### PR TITLE
test: [1600] Add tests for estimate gas with precompiles

### PR DESCRIPTION
**Pull request by nickeynikolovv**
This PR adds tests for estimate gas with Precompiles 

- Association and dissociation of accounts with tokens
- Some negative scenarios
- Transfer/Transfer from 
- Approve/allowance

- There are still more tests that have to be added afterwards

### CODE REVIEW

File: packages/server/tests/contracts/HederaTokenService.dbg.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "_format": "hh-sol-dbg-1",
+  "buildInfo": "../../../build-info/ffc708c862e1829402d865a511878cc8.json"
+}
```
Comment (AlfredoG87) [inline packages/server/tests/contracts/HederaTokenService.dbg.json:4]: do we need to commit this file?

File: packages/server/tests/contracts/EstimatePrecompileContract.dbg.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "_format": "hh-sol-dbg-1",
+  "buildInfo": "../../../build-info/a5263af18536fa5db5ec393e5db0fe0e.json"
+}
```
Comment (AlfredoG87) [inline packages/server/tests/contracts/EstimatePrecompileContract.dbg.json:4]: Also do we need this file?

File: packages/server/tests/acceptance/estimateGasPrecompile.spec.ts [hunk 3]
```diff
@@ -0,0 +1,896 @@
+/*-
+ *
+ * Hedera JSON RPC Relay
+ *
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
```
Comment (AlfredoG87) [inline packages/server/tests/acceptance/estimateGasPrecompile.spec.ts:5]: should we change this for `2023-2024`?

File: packages/server/tests/acceptance/estimateGasPrecompile.spec.ts [hunk 4]
```diff
@@ -0,0 +1,896 @@
...
+import { maxGasLimit } from '@hashgraph/json-rpc-relay/tests/helpers';
+import { create } from 'ts-node';
+
+describe.only('EstimatePrecompileContract tests', function () {
+  const signers: AliasAccount[] = [];
+  const prefix = '0x';
+  const CALL_EXCEPTION = 'CALL_EXCEPTION';
+  let contract: ethers.Contract;
+  let contractAccount1: ethers.Contract;
+  //  let contract1: ethers.Contract;
```
Comment (AlfredoG87) [inline packages/server/tests/acceptance/estimateGasPrecompile.spec.ts:39]: should we remove this?

File: packages/server/tests/acceptance/estimateGasPrecompile.spec.ts [hunk 5]
```diff
@@ -0,0 +1,896 @@
...
+      gasLimit: 500_000,
+    });
+  });
+
+  const baseGasCheck = (estimatedGasValue, expectedValue: number) => {
+    // handle deviation of 20%
+    expect(Number(estimatedGasValue)).to.be.lessThan(expectedValue * 1.4);
+  };
+
+  const isWithinDeviation = (
```
Comment (AlfredoG87) [inline packages/server/tests/acceptance/estimateGasPrecompile.spec.ts:249]: @ebadiere, is this similar to what you are working on for checking gas within range?

@nickeynikolovv The idea that I believe @ebadiere is working on, is to create a shared Util to do something similar across many other tests that struggle due to fluctuations on Gas used.

**Review by AlfredoG87 — COMMENTED** [review]
LG, mostly nits.

File: packages/server/tests/acceptance/estimateGasPrecompile.spec.ts [hunk 6]
```diff
@@ -0,0 +1,896 @@
...
+import { AliasAccount } from '../clients/servicesClient';
+import EstimatePrecompileContractJson from '../contracts/EstimatePrecompileContract.json';
+import ERC20MockJson from '../contracts/ERC20Mock.json';
+import Constants from '../../tests/helpers/constants';
+import RelayCalls from '../../../../packages/server/tests/helpers/constants';
+import ERC721MockJson from '../contracts/ERC721Mock.json';
+import { maxGasLimit } from '@hashgraph/json-rpc-relay/tests/helpers';
+import { create } from 'ts-node';
+
+describe.only('EstimatePrecompileContract tests', function () {
```
Comment (AlfredoG87) [inline packages/server/tests/acceptance/estimateGasPrecompile.spec.ts:33]: please remove this `.only`

Comment (nickeynikolovv) [reply packages/server/tests/acceptance/estimateGasPrecompile.spec.ts]: Done

File: packages/server/tests/acceptance/estimateGasPrecompile.spec.ts (hunk 4 above)
Comment (nickeynikolovv) [reply packages/server/tests/acceptance/estimateGasPrecompile.spec.ts]: Yes, done

File: packages/server/tests/acceptance/estimateGasPrecompile.spec.ts (hunk 3 above)
Comment (nickeynikolovv) [reply packages/server/tests/acceptance/estimateGasPrecompile.spec.ts]: Done

File: packages/server/tests/contracts/EstimatePrecompileContract.dbg.json (hunk 2 above)
Comment (nickeynikolovv) [reply packages/server/tests/contracts/EstimatePrecompileContract.dbg.json]: We don't need it, it is removed

File: packages/server/tests/contracts/HederaTokenService.dbg.json (hunk 1 above)
Comment (nickeynikolovv) [reply packages/server/tests/contracts/HederaTokenService.dbg.json]: We don't need it, it's removed

**Review by AlfredoG87 — APPROVED** [review]
Please Fix the DCO issue.
Otherwise looks good

**Pull request merged**